tests/integration/sync/test_sync_code.py

```diff
@@ -0,0 +1,206 @@
+import os
+import platform
+
+import logging
+import json
+import shutil
+import tempfile
+import time
+import uuid
+from pathlib import Path
+from unittest import skipIf
+
+import pytest
+import boto3
+
+from samcli.lib.utils.resources import (
+    AWS_APIGATEWAY_RESTAPI,
+    AWS_LAMBDA_FUNCTION,
+    AWS_STEPFUNCTIONS_STATEMACHINE,
+)
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+IS_WINDOWS = platform.system().lower() == "windows"
+# Some wait time for code updates to be reflected on each service
+LAMBDA_SLEEP = 3
+API_SLEEP = 5
+SFN_SLEEP = 5
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
```

Comment:
Can't we use some of these static vars from base class which is also defined there?